### DISCUSSION TITLE
Hosted Redis?

### DISCUSSION THREAD

**Discussion by bwalton** [sidekiq/sidekiq#5761] [Q&A] [answered]
Hello everyone,
What are some recommendations on hosted Redis providers? I've been through a few myself:

**Render.com**
Doesn't seem to work with the redis-client gem at all (at least for me). So while I thought the service itself was ok, I wouldn't be able to move to Sidekiq 7. I spoke with support (really nice) but never got to a resolution.

**Digital Ocean**
Worked alright for me minus the random bits of down time. I understand that there are maintenance windows but it would just stop accepting connections for a period of time completely stopping all of my workers. This only happened a few times but it was enough for me to run a Render Redis instance along side of it as a fail safe.

**Upstash**
This is who I'm currently trying out. I really like Upstash a lot but but I'm seeing tons of errors that I've never seen before on any other Redis host. It will work perfect one day and then the next it'll start with errors like:
- RedisClient::ReadTimeoutError
- RedisClient::CannotConnectError
- comparison of Integer with nil failed _(which looks like Sidekiq having issues connecting to Redis)_

Sidekiq will retry and everything is good but man, the constant errors aren't great.

I'm running on Heroku so my answer is _probably_ using them directly but I wanted to see what folks in here had to say.

**Comment by mperham** [comment, marked as answer]
Have you thought about running it yourself? The Redis docker image is really easy to run. Maybe someone else has experience running it on an EC2 instance or droplet.

IME if you can run Redis yourself, it's essentially bulletproof and will run for years without issue.

https://hub.docker.com/_/redis/

**Reply by bwalton** [reply]
That's probably the right answer at the end of the day, Mike. I'm just at the stage with my business where if something isn't the core of what I'm doing, I won't hesitate to pay someone else to do a better job.

Maybe in this case running a docker instance **would** be the better job.

**Reply by mperham** [reply]
Totally, it may not be right **for you** but it might be a viable option for others who see this discussion.

**Comment by collimarco** [comment]
I have been running Redis on a **DigitalOcean droplet** for years, without any issues. Why don't you do that? You have more control, better performance and lower costs.

**Reply by bwalton** [reply]
Yep, that's what I ended up doing and has been working great so far.
